src/Mvc/Mvc.ApiExplorer/src/EndpointMethodInfoApiDescriptionProvider.cs

```diff
@@ -0,0 +1,213 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Metadata;
+using Microsoft.AspNetCore.Mvc.Abstractions;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.AspNetCore.Mvc.ModelBinding.Metadata;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.AspNetCore.Routing.Patterns;
+using Microsoft.Extensions.Internal;
+
+namespace Microsoft.AspNetCore.Mvc.ApiExplorer
+{
+    internal class EndpointMethodInfoApiDescriptionProvider : IApiDescriptionProvider
+    {
+        private readonly EndpointDataSource _endpointDataSource;
+
+        // Executes before MVC's DefaultApiDescriptionProvider and GrpcHttpApiDescriptionProvider
+        // REVIEW: Does order matter here? Should this run after MVC?
+        public int Order => -1100;
+
+        public EndpointMethodInfoApiDescriptionProvider(EndpointDataSource endpointDataSource)
+        {
+            _endpointDataSource = endpointDataSource;
+        }
+
+        public void OnProvidersExecuting(ApiDescriptionProviderContext context)
+        {
+            foreach (var endpoint in _endpointDataSource.Endpoints)
+            {
+                if (endpoint is RouteEndpoint routeEndpoint
+                    && routeEndpoint.Metadata.GetMetadata<MethodInfo>() is { } methodInfo
+                    && routeEndpoint.Metadata.GetMetadata<IHttpMethodMetadata>() is { } httpMethodMetadata)
+                {
+                    // REVIEW: Should we add an ApiDescription for endpoints without IHttpMethodMetadata? Swagger doesn't handle
```

Comment:
Interesting q - MVC does add an entry for items that do not specify a method constraint (with a null value), so maybe it's fine to add one: https://github.com/dotnet/aspnetcore/blob/44bcd96cc93131bdcf184bd7e22364c7dab18a96/src/Mvc/Mvc.ApiExplorer/src/DefaultApiDescriptionProvider.cs#L341-L351

Reply:
I tried null initially since `ApiDescription.HttpMethod` is indeed nullable, but I get this when trying this with Swashbuckle 5.6.3.

```
fail: Microsoft.AspNetCore.Server.Kestrel[13]
      Swashbuckle.AspNetCore.SwaggerGen.SwaggerGeneratorException: Ambiguous HTTP method for action - . Actions require an explicit HttpMethod binding for Swagger/OpenAPI 3.0
         at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.GenerateOperations(IEnumerable`1 apiDescriptions, SchemaRepository schemaRepository) in Swashbuckle.AspNetCore.SwaggerGen.dll:token 0x60000f0+0x76
         at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.GeneratePaths(IEnumerable`1 apiDescriptions, SchemaRepository schemaRepository) in Swashbuckle.AspNetCore.SwaggerGen.dll:token 0x60000ef+0x60
         at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.GetSwagger(String documentName, String host, String basePath) in Swashbuckle.AspNetCore.SwaggerGen.dll:token 0x60000ed+0xe0
         at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider) in Swashbuckle.AspNetCore.Swagger.dll:token 0x6000009+0xa9
         at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpApplication`1 application) in Microsoft.AspNetCore.Server.Kestrel.Core.dll:token 0x6000a9f+0x1b8
```

Is there a workaround?